src/kernel/routing/NetZoneImpl.cpp

```diff
@@ -575,58 +466,192 @@ void NetZoneImpl::get_global_route(const NetPoint* src, const NetPoint* dst,
+{
+  /* Deal with the easy base case */
+  if (src->get_englobing_zone() == dst->get_englobing_zone()) {
+    *common_ancestor = src->get_englobing_zone();
+    *src_ancestor    = *common_ancestor;
+    *dst_ancestor    = *common_ancestor;
+    return;
+  }
+
+  const NetZoneImpl* src_as = src->get_englobing_zone();
+  const NetZoneImpl* dst_as = dst->get_englobing_zone();
+
+  xbt_enforce(src_as, "Host %s must be in a netzone", src->get_cname());
+  xbt_enforce(dst_as, "Host %s must be in a netzone", dst->get_cname());
+
+  /* (1) find the common parent.
+   * Before that, index_src and index_dst may be different, they both point to nullptr in path_src/path_dst
+   * So we move them down simultaneously as long as they point to the same content.
+   *
+   * This works because all SimGrid platform have a unique root element (that is the last element of both paths).
+   */
+  *src_ancestor       = nullptr;
+  *dst_ancestor       = nullptr;
+  *common_ancestor    = nullptr;
+  NetZoneImpl* parent = nullptr; // the netzone we dropped on the previous loop iteration
+  for (auto src_it = path_src->begin(), dst_it = path_dst->begin();
+       src_it != path_src->end() && dst_it != path_dst->end(); ++src_it, ++dst_it) {
+    if (*src_it != *dst_it) /* (2) we found the difference at least. Finalize the returned values */
+    {
+      *src_ancestor = *src_it; /* the first different parent of src */
+      *dst_ancestor = *dst_it; /* the first different parent of dst */
+      path_src->erase(path_src->begin(), src_it + 1);
+      path_dst->erase(path_dst->begin(), dst_it + 1);
+      break;
+    }
+
+    parent = *src_it;
+    // if we reach the end of one of the path (but not both), the corresponding NetPoint is in the netzone of the
+    // ancestor
+    if (src_it + 1 == path_src->end() && dst_it + 1 != path_dst->end()) {
+      *src_ancestor = *src_it;
+      *dst_ancestor = *(dst_it + 1);
+      path_dst->erase(path_dst->begin(), dst_it + 2);
+      break;
+    } else if (dst_it + 1 == path_dst->end() && src_it + 1 != path_src->end()) {
+      *src_ancestor = *(src_it + 1);
+      *dst_ancestor = *dst_it;
+      path_src->erase(path_src->begin(), src_it + 2);
+      break;
+    }
+  }
+
+  /* (2) check whether src and dst are in the same netzone */
+  if (*src_ancestor == *dst_ancestor) {
+    *common_ancestor = *src_ancestor;
+  } else {
+    xbt_enforce(parent != nullptr);
+    *common_ancestor = parent;
+  }
+
+  // maybe we should remove this check for performance reasons
+  xbt_enforce(parent != nullptr, "No common ancestor found for '%s' and '%s'", src->get_cname(), dst->get_cname());
+  xbt_enforce(*src_ancestor != nullptr, "No src ancestor found for '%s' and '%s'", src->get_cname(), dst->get_cname());
+  xbt_enforce(*dst_ancestor != nullptr, "No dst ancestor found for '%s' and '%s'", src->get_cname(), dst->get_cname());
+}
+
+/* Compute the route between a leaf NetPoint and an upper router */
```

Comment:
I have a hard time visualizing where in the hierarchy is that function used. Maybe an illustration or a more detailed comment would help.